### PR TITLE
Fixes locker stunlock exploit with slippery items

### PR DESCRIPTION
**Pull request by silicons**
Pushed me too far this time.

### CODE REVIEW

File: code/game/objects/items/devices/PDA/PDA_types.dm [hunk 1]
```diff
@@ -7,8 +7,11 @@
 	desc = "A portable microcomputer by Thinktronic Systems, LTD. The surface is coated with polytetrafluoroethylene and banana drippings."
 	ttone = "honk"
 
-/obj/item/device/pda/clown/Crossed(AM as mob|obj)
+/obj/item/device/pda/clown/Crossed(AM as mob|obj, oldloc)
 	if (istype(AM, /mob/living/carbon))
+		var/atom/A = oldloc
```
Comment (RemieRichards) [inline code/game/objects/items/devices/PDA/PDA_types.dm]: there's no need to cast, just:
```
if(get_turf(oldloc) == get_turf(src))
    return 
```

**Comment by QualityVan** [comment]
i got griffed
pls nerf

**Comment by Repukan** [comment]
>Grudge coding

You're not a mantainer, you can't get away with this

**Comment by RemieRichards** [comment]
He can, as this is something we want fixed, just too busy/lazy with other things to get fixed.

Personally I think the check should be done once, in ```forceMove()``` (ie: if it's the same turf, don't call ```Entered()```/```Crossed()```, etc, since the default byond calls to those procs make it IMPOSSIBLE to be in the same tile and call them.

if you keep them in ```Crossed()``` overrides you're bound to miss one/future expansion.

**Comment by Repukan** [comment]
It's joke

**Comment by RemieRichards** [comment]
ur a joke

**Comment by Repukan** [comment]
No u

**Comment by kevinz000** [comment]
@RemieRichards so you're saying I should tamper with forceMove() to not call crossed/entered/exited/whatever if the destination is the same as the source?

**Comment by kevinz000** [comment]
wait but the issue is lockers dumping everything onto the floor, so if I did that wouldn't it break functionality if someone had a subtype of closets that utilize Crossed/Entered?

**Comment by RemieRichards** [comment]
You still DO the movement (that's the one allowed ```loc = ``` in the codebase, inside ```forceMove()```) you just don't call ```Entered()```/```Crossed()```, etc. (and ```Exited()``` on the  ''''old'''' loc if force move does that)

**Review by Cyberboss — DISMISSED** [review]
Do it in a generic way

**Comment by Cyberboss** [comment]
I personally think Crossed is the only thing that shouldn't be called if the turfs are the same

**Comment by kevinz000** [comment]
ok

**Comment by Cyberboss** [comment]
kev

**Comment by Repukan** [comment]
If you can figure out and fix out why people get multiple limb damage when a locker is drilled or attacked by swarmers or pulse rifles, you can close like 4 issues at once, dood.

**Comment by kevinz000** [comment]
ouch

**Comment by kevinz000** [comment]
fixed sir.

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -250,15 +250,16 @@
 			destination.Entered(src, oldloc)
 			if(destarea && old_area != destarea)
 				destarea.Entered(src, oldloc)
-
-			for(var/atom/movable/AM in destination)
-				if(AM == src)
-					continue
-				AM.Crossed(src, oldloc)
+			var/sameturf = get_turf(oldloc) == get_turf(destination)
```
Comment (lzimann) [inline code/game/atoms_movable.dm]: why is this a var and not just `if(get_turf(oldloc) != get_turf(destination)` ?

**Comment by RemieRichards** [comment]
I disagree with @Cyberboss You shouldn't call any of the byond in built "movement" functions if it's the same turf, as there's no "movement" occuring.

Either we:
A. Consider it as moving from the locker to the turf, which means ALL of byond's stuff is valid (including the Crossed "bug")
or
B. Consider it as you already being on that turf, so none of it should be called.

We should not be picking and choosing parts of the movement ""system"" like that, it's inconsistent.

**Comment by AnturK** [comment]
I kinda agree with Remie here. The atom -> turf it's on movement isn't that rare so i believe it should be treatead as any other movement. Not sure on the solution on locker slip though.

**Comment by zamolxius** [comment]
emergent feature IMHO

**Comment by optimumtact** [comment]
There is currently a 2-1 maintainer majority on implementing it how remie suggests, so please do that.

**Comment by KorPhaeron** [comment]
Stale

**Pull request closed without merging**